runner/src/internal.rs

```diff
@@ -288,8 +301,9 @@ pub async fn run_step(context: &Context, step: Step) -> HashSet<StatusResultValu
                 "{} ⊢ (finished) {}",
                 context.prefix, background_status.value
             );
+
             if (background_status.value == StatusResultValue::Error
-                || values.contains(&StatusResultValue::Error)
+                || foreground_result.values.contains(&StatusResultValue::Error)
                 || context.opt.logs)
                 && !logs.is_empty()
```

Comment:
Is this check needed for pushing the step to `step_result.failed_steps_prefixes`? Or should `step_result.failed_steps_prefixes` be updated regardless of the content of `logs`?

Reply:
Done